server/src/main/java/org/elasticsearch/plugins/internal/DocumentParsingProvider.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugins.internal;
+
+/**
+ * An interface to provide instances of document parsing observer and reporter
+ */
+public interface DocumentParsingProvider {
+    DocumentParsingProvider EMPTY_INSTANCE = new DocumentParsingProvider() {
+        @Override
+        public DocumentSizeObserver newDocumentSizeObserver() {
+            return DocumentSizeObserver.EMPTY_INSTANCE;
+        }
+
+        @Override
+        public DocumentParsingReporter getDocumentParsingReporter() {
+            return DocumentParsingReporter.EMPTY_INSTANCE;
+        }
+
+        @Override
+        public DocumentSizeObserver newDocumentSizeObserver(long normalisedBytesParsed) {
+            return DocumentSizeObserver.EMPTY_INSTANCE;
+        }
+    };
+
+    /**
+     * @return a new 'empty' observer to use when observing parsing
+     */
+    DocumentSizeObserver newDocumentSizeObserver();
+
+    /**
+     * @return an observer to use when continue observing parsing based on previous result
+     */
+    DocumentSizeObserver newDocumentSizeObserver(long normalisedBytesParsed);
```

Comment:
Can we use a more distinctive name from creating a "real" observer above? eg `newFixedDocumentSizeObserver`, and make it clear in the javadoc this observer does not actually observe anything, it reports the exact size passed in. The javadoc currently implies it would keep increasing the size when passed to a parser.

Reply:
it is not always a fixedDcoumetnSizeObserver.
If the parsing has not happened in the `IngestService` we still want to create a new Metering DocumetnSizeObserver.
I will  keep the method name, but will remove the javadoc wording indicating that it will continue parsing (this was based on the previous idea that we wanted to keep the interfaces in ES independent of the implementation in serverless)
I don't want to make this javadoc too detailed, as it would basically mean I would have to document serverless plugin here

Reply:
I renamed the 2 methods as per your suggestion there and updated the javadoc